### PR TITLE
[FEA] Raft RNG updated API

### PR DESCRIPTION
**Pull request by MatthiasKohl**
This PR updates usage of RAFT RNG functionality throughout the cugraph code-base, including using the new RAFT RngState to pass state between cugraph and cugraph-ops.
Finally, I removed unnecessary includes from `raft/random` where applicable.

### CODE REVIEW

File: cpp/src/sampling/neighborhood.cu
Comment (kaatish) [inline cpp/src/sampling/neighborhood.cu:31]: You will have to make the corresponding changes to cpp/include/cugraph/algorithms.hpp file. The functions sample_neighbors* do not have a python binding yet so CI should not break because of this.
@rlratzel thoghts?

Comment (MatthiasKohl) [reply cpp/src/sampling/neighborhood.cu:31]: Thanks for catching this. I've updated the public API in algorithms.
I don't see any uses in python or binding code, so I think this should be good enough for now.

Comment (rlratzel) [reply cpp/src/sampling/neighborhood.cu:31]: > You will have to make the corresponding changes to cpp/include/cugraph/algorithms.hpp file. The functions sample_neighbors* do not have a python binding yet so CI should not break because of this. @rlratzel thoghts?

This should be okay since the python binding for neighborhood sampling will use the C API, which does not expose this function signature to the binding code.

Comment (ChuckHastings) [reply cpp/src/sampling/neighborhood.cu:31]: I will merge this change (once it is merged into the code base) into my refactor branch for uniform neighborhood sampling.  I've been talking with Corey about wanting to make this change, so I'm ready.

**Review by ChuckHastings — APPROVED** [review]
Do we have an issue somewhere to track the issue we were seeing with the PC Generator (the default for raft)?

**Comment by MatthiasKohl** [comment]
I'll open an issue

**Comment by MatthiasKohl** [comment]
I've opened issue #2266 for this and the tests here seem to be passing.
@kaatish can we merge this PR?

**Review by kaatish — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**
